### PR TITLE
Feature: Copilot Chat

### PR DESCRIPTION
**Pull request by adrianwyatt**
### Motivation and Context
- Adds frontend webapp (React)
- Add backend SKWebApi (SK REST API Service) for CopilotChat sample
- Adds CosmosDB connector memory store

### Description

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ var kernel = Kernel.Builder.Build();
 
 // For Azure Open AI details please see
 // https://learn.microsoft.com/azure/cognitive-services/openai/quickstart?pivots=rest-api
-kernel.Config.AddAzureOpenAITextCompletion(
+kernel.Config.AddAzureOpenAITextCompletionService(
```
Comment (lemillermicrosoft) [inline README.md:107]: nit: we shouldn't update this yet until nuget is spun

File: samples/dotnet/FileCompression/FileCompressionSkill.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@
 using Microsoft.SemanticKernel.Orchestration;
 using Microsoft.SemanticKernel.SkillDefinition;
 
-namespace FileCompression;
+namespace Microsoft.SemanticKernel.Skills.FileCompression;
```
Comment (lemillermicrosoft) [inline samples/dotnet/FileCompression/FileCompressionSkill.cs:12]: I think this file needs to be reverted.

Comment (lemillermicrosoft) [reply samples/dotnet/FileCompression/FileCompressionSkill.cs:12]: And others in this folder.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryRecord.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Skills.Memory.CosmosDB;
+
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryRecord.cs:13]: Does this need to be public? If so, xml docs.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryRecord.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Newtonsoft.Json;
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryRecord.cs:9]: Are these required?

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+namespace Skills.Memory.CosmosDB;
+
+/// <summary>
+/// An implementation of <see cref="IMemoryStore{TEmbedding}"/> for Azure Cosmos DB.
+/// </summary>
+/// <remarks>The Embedding data is saved to the Azure Cosmos DB database container specified in the constructor.
+/// The embedding data persists between subsequent instances and has similarity search capability, handled by the client as Azure Cosmos DB is not a vector-native DB.
+/// </remarks>
+public class CosmosDBMemoryStore<TEmbedding> : IMemoryStore<TEmbedding>, IDisposable
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs:27]: Is this being used at all yet? Would be nice to have tests.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    {
+        var container = this._client.GetContainer(this._databaseName, this._containerName);
+
+        return container.DeleteItemAsync<CosmosDBMemoryRecord>(
+            this._toCosmosFriendlyId(key),
+            new Microsoft.Azure.Cosmos.PartitionKey(collection),
+            cancellationToken: cancel);
+    }
+
+    private string _toCosmosFriendlyId(string id)
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs:207]: I'm surprised our rules didn't flag this. I'd expect `ToCosmosFriendlyId`

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs (hunk 5 above)
Comment (SergeyMenshykh) [inline dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs:27]: Does it need to implement IDisposable? I don't see any resource the class create that need to be released/freed.

Comment (craigomatic) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs:27]: We should drop this - I believe this was a carryover from a prior implementation. Both the interface and the Dispose() implementation are not necessary.

Comment (adrianwyatt) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs:27]: I'll make the change here - standby.

Comment (adrianwyatt) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs:27]: https://github.com/microsoft/semantic-kernel/pull/336

File: README.md (hunk 1 above)
Comment (adrianwyatt) [reply README.md:105]: Looks like we spun out a NuGEt.

File: samples/dotnet/FileCompression/FileCompressionSkill.cs (hunk 2 above)
Comment (adrianwyatt) [reply samples/dotnet/FileCompression/FileCompressionSkill.cs:12]: will do.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryRecord.cs (hunk 4 above)
Comment (adrianwyatt) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryRecord.cs:9]: Removed.

File: dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs (hunk 5 above)
Comment (adrianwyatt) [reply dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosDBMemoryStore.cs]: Not yet - tests will be coming shortly.

**Comment by adrianwyatt** [comment]
Abandoning to resolve conflicts in a non-feature branch.

**Pull request closed without merging**